src/Cake.Common/Tools/MSBuild/MSBuildSettings.cs

```diff
@@ -281,5 +293,13 @@ private string GetPropertyValueOrDefault(string propertyName, string @default =
             var propertyValue = string.Join(";", propertyValues);
             return propertyValue;
         }
+
+        private void SetTargets(string value)
+        {
+            foreach (var target in value.Split(";").Where(p => !string.IsNullOrEmpty(p)))
```

Comment:
I would expect something like this
```suggestion
            Targets.Clear();
            foreach (var target in value?.Split(';', _targetStringSplitOptions) ?? Array.Empty<string>())
```
so setting `Target` to a new value isn't keeping the old values or what do you think @augustoproiete ?

Where is something like this on top of class
```csharp
#if NETCOREAPP3_1
        private static readonly StringSplitOptions _targetStringSplitOptions = StringSplitOptions.RemoveEmptyEntries;
#else
        private static readonly StringSplitOptions _targetStringSplitOptions = StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries;
#endif
```